transport/grpc/config.go

```diff
@@ -255,6 +259,7 @@ func (t *transportSpec) buildOutbound(outboundConfig *OutboundConfig, tr transpo
 		return nil, newTransportCastError(tr)
 	}
 
+	RegisterCompressors(kit.Compressors()...)
```

Comment:
I am not sure if this is the right place if we want to call RegisterCompressors only once in total. Is there a better place?

Reply:
This may be a good concede that we’ve been defeated by Go’s global scope dependency injection system and roll with the punch.

I’ll have to talk with @abhinav and @prashantv about the right course here, but I suspect the answer is that we thread the compressor name or compressor instance through the outbound configuration, and that the compressor has to also be registered in an `init()` with gRPC’s registration mechanism.

Ideally we do this in a way that would be forward compatible if we scuttled grpc-go in favor of x/net/http2 some day in the future. For that to work, we may need to “fake thread” the dependency through the kit even though it’s also been registered. The purpose of faking it would be to enforce that the user has passed the Compressor through a Configurator constructor option. That would allow us in the future to thread the dependency properly.

That all might be irrelevant though, since the avenue for introducing an http2 transport is just creating a new transport package and probably also thread dependencies through a yarpchttp2fx module.

If you’ll forgive the stream of consciousness, I think the least-worst solution here is to skip threading the Compressor through the kit. We would just expose the gRPC Outbound Compressor option and Config, then register the compressor using gRPC’s API directly.

Reply:
If these get registered multiple times, the last registration with a given name will survive, which is OK for us, I think, but it is not OK if we try to do the registrations in parallel.

In some sense, it may be OK, to just do what you suggested in the last paragraph, because GRPC already has some compressors, that can be easily [(?)](https://github.com/grpc/grpc-go/blob/master/encoding/gzip/gzip.go#L38) registered.